bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -402,7 +402,12 @@ class PlotCanvasView extends Renderer.View
     # reach the frame in time (sometimes) so force an update here for now
     @model.get('frame')._update_mappers()
 
+    # Get context and restore to its initial state. The restoring
+    # Would stricly speaking not be necessary, but in case there is an error
+    # during drawing, restore() won't be called enough. This guards for that.
     ctx = @canvas_view.ctx
+    for i in [0..3]
+      ctx.restore()
```

Comment:
I'm a little confused by this, can you elaborate more? This seems like it will unconditionally restore, regardless of whether there was an error or not. If not, in which case it will unexpectedly blow away any context `push` states? Or are you assuming / relying on the context state to otherwise be "zeroed" before getting to the code that might error out?